From 69a85af0f84be53eed682db45ff97ddf651cae2d Mon Sep 17 00:00:00 2001
From: Chris Fiege <cfi@pengutronix.de>
Date: Thu, 20 Apr 2023 23:04:55 +0200
Subject: [PATCH 19/38] gluon-status-page: Fix redirect to actual IPv4 addr

Finding the actual IPv4 address of the node relied on "-br" of ip.
Since Gluon v2022 we must live with busybox ip...
This change works around this change...
---
 .../luasrc/lib/gluon/status-page/controller/status-page.lua     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/gluon-status-page/luasrc/lib/gluon/status-page/controller/status-page.lua b/package/gluon-status-page/luasrc/lib/gluon/status-page/controller/status-page.lua
index 56463d79..ab394494 100644
--- a/package/gluon-status-page/luasrc/lib/gluon/status-page/controller/status-page.lua
+++ b/package/gluon-status-page/luasrc/lib/gluon/status-page/controller/status-page.lua
@@ -84,7 +84,7 @@ entry({}, call(function(http, renderer)
 		if match(node_ip, parse_ip(site.next_node.ip4()), 4) then
 			-- The user has visited the status page via the ipv4 next-node address
 			-- Redirect the user to our unique ipv4 address to avoid switching nodes
-			local process = io.popen('ip -br -4 address show br-client', 'r')
+			local process = io.popen('ip -4 a show dev br-client','r')
 			if process then
 				local output = process:read('*a')
 				process:close()
-- 
2.34.1

